sources/unstructured_data/inbox/__init__.py

```diff
@@ -0,0 +1,133 @@
+) -> TDataItem:
+
+    last_date = initial_date.last_value.strftime("%d-%b-%Y")
+
+    def read_messages(client_: imaplib.IMAP4_SSL, criteria_: Sequence[Any]):
+        _, messages = client_.search(None, *criteria_)
+        message_ids = messages[0].split()
+
+        if not message_ids:
+            logger.warning("No emails found.")
+
+        for message_id in message_ids:
+            _, data = client_.fetch(message_id, "(RFC822)")
+            raw_email = data[0][1]
+            msg = email.message_from_bytes(raw_email)
+            email_data = {
+                'message_uid': message_id,
+                'message_id': msg['Message-ID'],
+                'from': msg['From'],
+                'subject': msg['Subject'],
+                'date': pendulum.parse(msg['Date'], strict=False),
+                'content_type': msg.get_content_type(),
+                'body': get_email_body(msg),
+            }
+            yield email_data
+
+    with imaplib.IMAP4_SSL(credentials["host"]) as client:
+        client.login(credentials["username"], credentials["password"])
+        client.select(folder, readonly=True)
+        criteria = [f'(SENTSINCE {last_date})']
+
+        if filter_emails:
+            logger.info(f"Load emails only from: {filter_emails}")
+
+            for email_ in filter_emails:
+                criteria.extend([f'(FROM {email_})'])
+                yield from read_messages(client, criteria)
+        else:
+            yield from read_messages(client, criteria)
+
+
+@dlt.transformer(name="attachments", write_disposition="replace")
+def get_attachments_by_uid(
+    item: TDataItem,
+    storage_folder_path: str,
+    credentials: Dict[str, str] = dlt.secrets.value
+) -> TDataItem:
+    message_uid = item["message_uid"]
+    with imaplib.IMAP4_SSL(credentials["host"]) as client:
+        client.login(credentials["username"], credentials["password"])
+        client.select()
+
+        response, data = client.uid('fetch', message_uid, '(RFC822)')
+
+        if response == 'OK':
+            raw_email = data[0]
+            if raw_email:
+                raw_email = data[0][1]
+                msg = email.message_from_bytes(raw_email)
+
+                for part in msg.walk():
+                    content_disposition = part.get("Content-Disposition", "")
+                    if "attachment" in content_disposition:
+                        filename = part.get_filename()
+                        if filename:
+                            attachment_data = part.get_payload(decode=True)
+                            attachment_path = os.path.join(storage_folder_path, message_uid.decode("utf-8")  + filename)
+                            os.makedirs(os.path.dirname(attachment_path), exist_ok=True)
+
+                            with open(attachment_path, 'wb') as f:
+                                f.write(attachment_data)
+
+                            item.update({"file_name": filename, "file_path": os.path.abspath(attachment_path)})
```

Comment:
clone item first with copy(). why? because you are yielding and then overwriting the file name. and this is still the same dict - if you have many attachments per email
also please add content type!